flink-streaming-java/src/main/java/org/apache/flink/streaming/api/operators/InternalWatermarkCallbackService.java

```diff
@@ -153,6 +153,7 @@ public void invokeOnWatermarkCallback(Watermark watermark) throws IOException {
 				}
 			}
 		}
+		cleanupRegisteredKeys();
```

Comment:
Yes unfortunately it is just for testing.  I would like to get rid of it because it implies an additional iteration. You can comment it and see which test fails in the `CEPOperatorsTest`.